backend/inspirehep/records/marshmallow/literature/utils.py

```diff
@@ -20,3 +28,33 @@ def get_parent_record(data):
         data, "publication_info.parent_record"
     )
     return next(book_records, {})
+
+
+def latex_encode(text, contains_math=False):
+    """Encode a string for use in a LaTeX format.
+
+    Args:
+        contains_math (bool): when True, math environments delimited by $...$
+        or \\(...\\) are preserved to avoid double escaping.
+    """
+    if text is None:
+        return None
+
+    encode = UnicodeToLatexEncoder(
+        replacement_latex_protection="braces-after-macro"
+    ).unicode_to_latex
+
+    if not (contains_math and ("$" in text or r"\(" in text)):
+        return encode(text)
+
+    math_expressions = MATH_EXPRESSION_REGEX.findall(text)
+
+    encoded_math_expressions = {
+        encode(expression): expression for expression in math_expressions
+    }
+    encoded_text = encode(text)
+
+    for encoded_math, math in encoded_math_expressions.items():
+        encoded_text = encoded_text.replace(encoded_math, math)
+
```

Comment:
interresting approach to encode everything and then substitute parts of the original back

how about avoiding encoding math parts by partitioning the text string and then recombining?

```python

parts = re.split(MATH_EXPRESSION_REGEX, title)                                                                                                   

newtitle = ''.join([encode(t) if i % 2 == 0 else t for i, t in enumerate(parts)])

```

Reply:
I thought it wouldn't work so easily because you don't know whether the string starts with math mode, but after testing it seems you get a leading empty string in this case:
```
In [5]: MATH_EXPRESSION_REGEX.split("$math$ non-math $math$")                                                                                                                                                        
Out[5]: ['', '$math$', ' non-math ', '$math$', '']
```
So good suggestion :+1: